client/components/users/user-form.js

```diff
@@ -1,85 +1,164 @@
-import React from 'react'
-import {updateUser} from '../../store/user'
+import React, {useState, useEffect} from 'react'
+import {useSelector, useDispatch} from 'react-redux'
 import {fetchAllSkills} from '../../store/skill'
-import {connect} from 'react-redux'
-import {withRouter} from 'react-router'
+import {updateUser} from '../../store/user'
+import Container from 'react-bootstrap/Container'
+import Card from 'react-bootstrap/Card'
+import Media from 'react-bootstrap/Media'
+import Image from 'react-bootstrap/Image'
 import Form from 'react-bootstrap/Form'
+import Accordion from 'react-bootstrap/Accordion'
 import Button from 'react-bootstrap/Button'
-import Container from 'react-bootstrap/Container'
-import SkillsSelect from '../skills/skills-select'
 
-class UserForm extends React.Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      email: this.props.email,
-      location: this.props.userLocation ? this.props.userLocation : '',
-      skillSelection: 0
-    }
+export default function UserForm({history}) {
+  const dispatch = useDispatch()
 
-    this.handleSubmit = this.handleSubmit.bind(this)
-    this.handleChange = this.handleChange.bind(this)
-    this.handleSelection = this.handleSelection.bind(this)
-  }
+  useEffect(() => {
+    dispatch(fetchAllSkills())
+  }, [])
 
-  componentDidMount() {
-    this.props.fetchAllSkills()
-  }
+  const user = useSelector(state => state.user)
+  const skills = useSelector(state => state.skills)
+  const [inputs, setInputs] = useState({
+    photoURL: '',
+    location: ''
+  })
+  const [skillIds, setSkillIds] = useState([])
 
-  handleSubmit(evt) {
-    evt.preventDefault()
-    this.props.updateUserStore({...this.state}, this.props.userId)
-    this.props.history.push('/home')
-  }
+  useEffect(
+    () => {
+      if (user.skills && user.skills.length) {
+        setSkillIds(
+          user.skills.reduce(
+            (skillsArr, currSkill) => [...skillsArr, currSkill.id],
+            []
+          )
+        )
+      }
+    },
+    [user]
+  )
 
-  handleChange(evt) {
-    evt.preventDefault()
-    this.setState({[evt.target.id]: evt.target.value})
+  const handleInput = e => {
+    setInputs({
+      ...inputs,
+      [e.target.id]: e.target.value
+    })
   }
 
-  handleSelection(evt) {
-    this.setState({skillSelection: evt.target.value})
+  const handleSwitch = (e, skill) => {
+    const id = parseInt(e.target.id, 10)
+    if (skillIds.includes(id)) {
+      user.skills = user.skills.filter(userSkill => skill.id !== userSkill.id)
+      setSkillIds(skillIds.filter(skillId => skillId !== id))
```

Comment:
line 52/53 can be cleaned up with `user.skills.filter(({ id }) => skill.id !== id)` or similar syntax. Not a requirement, but just an idea..

Also, the `filter` code shared between 52 & 53 are very similar. maybe we can abstract this filter logic to be a function that can take these as inputs so we can be more DRY here... also not a requirement. might not be worth it

Reply:
This is the portion where I mentioned that it really ought to have its own reducer; that's exactly the kind of code you'd find in a reducer, anyway. I'm looking into defining our existing junction table concretely so we have direct access to the ids. Then, actually, all of the logic pertaining to parsing out any ids can be removed entirely.